vpr/src/timing/timing_graph_builder.cpp

```diff
@@ -1,3 +1,196 @@
+ *                   ^                ^
+ *                   |                |
+ *                   +-----+    +-----+
+ *                         |    |
+ *                       +--------+
+ *                   --->| CPIN   |
+ *                       +--------+
+ *
+ * Where 'SINK B' represents the data input of sequential element 'B' (e.g. 
+ * Flip-Flop D pin), 'SRC B' the data output (SOURCE) of sequential element 
+ * 'B' (e.g. Flip-Flop Q pin), and 'CPIN' the clock input pin of sequential 
+ * element 'B'.
+ *
+ *
+ * Following the above transformations, we arrive at the following timing graph 
+ * structure, which corresponds to the architectural primitive described above:
+ *
+ *       +--------+                                             +--------+
+ *   --->| IPIN e |-------------------------------------------->| OPIN g |--->
+ *       +--------+                                             +--------+
+ *            |
+ *            +------------------------------------+
+ *                                                 |
+ *                                                 v
+ *       +--------+        +--------+         +--------+        +--------+
+ *   --->| SINK A |        | SRC A  |-------->| SINK B |        | SRC B  |--->
+ *       +--------+        +--------+         +--------+        +--------+
+ *            ^                ^                   ^                ^
+ *            |                |                   |                |
+ *         +--+                |                   |                |
+ *         |      +------------+                   |                |
+ *         |      |                                |                |
+ *       +----------+                              |                |
+ *       |          |------------------------------+                |
+ *   --->| CPIN clk |                                               |
+ *       |          |-----------------------------------------------+
+ *       +----------+                        
+ *
+ * Note that we have also used only a single CPIN for both 'A' and 'B' (since 
+ * they are controlled by the same clock).
+ *
+ *
+ *
+ * Building the Timing Graph From VPR's Data Structures
+ * ----------------------------------------------------
+ *
+ * VPR does not directly model the intenals of netlist primitives (e.g. internal
+ * sequential elements like 'A' or 'B' above). Instead, various attributes are 
+ * tagged on the pins of the primitive which indicate:
+ *     - whether a pin is sequential, combinational or a clock
+ *     - whether the pin is combinationally connected to another pin 
+ *       within the primitive.
+ * 
+ * Mostly there is a one-to-one correspondance between netlist pins and tnodes,
+ * the only exception is for sequential-sequential connections within a primtive
+ * (e.g. the fully internal 'A' to 'B' timing path above).
+ *
+ * As a result we make a distinction between tnodes which are strictly "internal"
```

Comment:
What is the modelling affect of internal vs external?  Or is simply a label, and has no affect on the timing analysis results?  Please add answer to comment.